src/compiler/checker.ts

```diff
@@ -6609,16 +6608,18 @@ namespace ts {
         }
 
         function getTypeReferenceType(node: TypeReferenceNode | ExpressionWithTypeArguments | JSDocTypeReference, symbol: Symbol) {
+            const typeArguments = typeArgumentsFromTypeReferenceNode(node); // Do unconditionally so we mark type arguments as referenced.

```

Comment:
`getTypeFromTypeNode` dispatches to a bunch of functions that are all cached, so the only danger is an arbitrarily long list of type arguments. Either way is fine with me, but I'd prefer to have some measurement that says the duplicate way is measurably slower.